### PR TITLE
chore(deps): Bump operator-rs to 0.67

### PR DESCRIPTION
**Pull request by labrenbe**
# Description

Bump operator-rs to 0.67 & other dependencies

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] CRD documentation for all fields, following the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
- [ ] Changes need to be "offline" compatible
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] Code contains useful logging statements
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated. Follows the [style guide](https://docs.stackable.tech/home/nightly/contributor/docs-style-guide).
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
- [ ] [Roadmap](https://github.com/orgs/stackabletech/projects/25/views/1) has been updated
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,18 @@ All notable changes to this project will be documented in this file.
...
+
+- `operator-rs` `0.64` -> `0.67` ([#571]).
+- `axum` `0.6` -> `0.7` ([#571]).
+- `built` `0.6` -> `0.7` ([#571]).
+- `hyper` `0.14` -> `1.3` ([#571]).
+- `reqwest` `0.11` -> `0.12` ([#571]).
+- `snafu` `0.7` -> `0.8` ([#571]).
+- `strum` `0.25` -> `0.26` ([#571]).
+
+[#465]: https://github.com/stackabletech/opa-operator/pull/571
```
Comment (maltesander) [inline CHANGELOG.md:17]: We agreed on not mentioning dependency upgrades a while ago, i think you can leave this out.
```suggestion
[#571]: https://github.com/stackabletech/opa-operator/pull/571
```

File: Cargo.toml [hunk 2]
```diff
@@ -10,24 +10,24 @@ edition = "2021"
...
 semver = "1.0"
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
-snafu = "0.7"
-stackable-operator = { git = "https://github.com/stackabletech/operator-rs.git", tag = "0.64.0" }
-strum = { version = "0.25", features = ["derive"] }
+snafu = "0.8"
+stackable-operator = { git = "https://github.com/stackabletech/operator-rs.git", tag = "stackable-operator-0.67.0" }
+strum = { version = "0.26", features = ["derive"] }
 tokio = { version = "1.29", features = ["full"] }
```
Comment (maltesander) [inline Cargo.toml:31]:
```suggestion
strum = { version = "0.26", features = ["derive"] }
tokio = { version = "1.37", features = ["full"] }
```

File: Cargo.toml [hunk 3]
```diff
@@ -10,24 +10,24 @@ edition = "2021"
 repository = "https://github.com/stackabletech/opa-operator"
 
 [workspace.dependencies]
-axum = "0.6"
-built = { version = "0.6", features = ["chrono", "git2"] }
+axum = "0.7"
+built = { version ="0.7", features = ["chrono", "git2"] }
 clap = "4.3"
```
Comment (maltesander) [inline Cargo.toml:15]: Clap should not be breaking
```suggestion
built = { version ="0.7", features = ["chrono", "git2"] }
clap = "4.5"
```

File: rust/crd/src/lib.rs [hunk 4]
```diff
@@ -15,7 +15,7 @@ use stackable_operator::{
     config::{fragment, fragment::Fragment, fragment::ValidationError, merge::Merge},
     k8s_openapi::apimachinery::pkg::api::resource::Quantity,
     kube::CustomResource,
-    product_config_utils::{ConfigError, Configuration},
+    product_config_utils::Configuration,
```
Comment (maltesander) [inline rust/crd/src/lib.rs:18]: Optional: You could do something like this and not change the error types further down
```
    product_config_utils::{Configuration, Error as ConfigError},
```

File: rust/operator-binary/src/controller.rs [hunk 5]
```diff
@@ -14,9 +14,9 @@ use stackable_opa_crd::{
 };
 use stackable_operator::{
     builder::{
-        resources::ResourceRequirementsBuilder, ConfigMapBuilder, ContainerBuilder,
-        FieldPathEnvVar, ObjectMetaBuilder, ObjectMetaBuilderError, PodBuilder,
-        PodSecurityContextBuilder, VolumeBuilder,
+        configmap::ConfigMapBuilder, meta::ObjectMetaBuilder, pod::container::ContainerBuilder,
+        pod::container::FieldPathEnvVar, pod::resources::ResourceRequirementsBuilder,
+        pod::security::PodSecurityContextBuilder, pod::volume::VolumeBuilder, pod::PodBuilder,
```
Comment (maltesander) [inline rust/operator-binary/src/controller.rs:19]: Can we summarize this the deps under `pod::`, `pod::container::` etc.?

File: rust/operator-binary/src/main.rs [hunk 6]
```diff
@@ -55,11 +73,11 @@ struct OpaRun {
 }
 
 #[tokio::main]
-async fn main() -> Result<(), error::Error> {
+async fn main() -> Result<(), Error> {
```
Comment (maltesander) [inline rust/operator-binary/src/main.rs:76]: We can go for anyhow here and remove the Error enum.
```suggestion
async fn main() -> anyhow::Result<()> {
```

File: rust/operator-binary/src/main.rs [hunk 7]
```diff
@@ -34,6 +34,24 @@ pub mod built_info {
     pub const CARGO_PKG_VERSION: &str = env!("CARGO_PKG_VERSION");
 }
 
+#[derive(Debug, Snafu)]
```
Comment (maltesander) [inline rust/operator-binary/src/main.rs:37]: Cannot comment on the code line but above should be 
```
pub mod built_info {
    include!(concat!(env!("OUT_DIR"), "/built.rs"));
}
```
and adapt e.g. `CARGO_PKG_VERSION` to `PKG_VERSION` (see main file here https://github.com/stackabletech/listener-operator/pull/183/files)

**Review by maltesander — CHANGES_REQUESTED** [review]

File: Cargo.toml [hunk 8]
```diff
@@ -10,24 +10,24 @@ edition = "2021"
...
-reqwest = { version = "0.11", features = ["json"] }
+reqwest = { version ="0.12", features = ["json"] }
 semver = "1.0"
 serde = { version = "1.0", features = ["derive"] }
 serde_json = "1.0"
-snafu = "0.7"
-stackable-operator = { git = "https://github.com/stackabletech/operator-rs.git", tag = "0.64.0" }
-strum = { version = "0.25", features = ["derive"] }
+snafu = "0.8"
+stackable-operator = { git = "https://github.com/stackabletech/operator-rs.git", tag = "stackable-operator-0.67.0" }
```
Comment (sbernauer) [inline Cargo.toml:29]: Sorry, there is already a new version available
```suggestion
stackable-operator = { git = "https://github.com/stackabletech/operator-rs.git", tag = "stackable-operator-0.67.1" }
```

File: rust/crd/src/lib.rs (hunk 4 above)
Comment (labrenbe) [reply rust/crd/src/lib.rs:18]: I see what you mean but I would prefer to not use the `ConfigError` name anymore.

**Review by maltesander — APPROVED** [review]
LGTM!

**Pull request merged**